lib/ansible/modules/packaging/language/pip.py

```diff
@@ -472,30 +473,37 @@ def setup_virtualenv(module, env, chdir, out, err):
         module.exit_json(changed=True)
 
     cmd = module.params['virtualenv_command']
+    cmd_args = ""
+
+    cmd_array = shlex.split(module.params['virtualenv_command'])
```

Comment:
no need to go through all this, just pass a cmd and args as a list to run cmd and shlex quoting already happens there.

Reply:
Please see my comments above - I tried it and got:

```
TypeError: expected str, bytes or os.PathLike object, not list
```

Reply:
Ah, figured out the problem and fixed it.